src/collector/Storage.cpp

```diff
@@ -146,6 +146,23 @@ void Storage::handle_backend(Backend & backend)
         it->second.add_backend(backend);
     }
 
+    if (backend.group_changed()) {
+        int old_id = backend.get_old_group_id();
+        auto it_old = m_groups.find(old_id);
+        if (it_old != m_groups.end()) {
+            BH_LOG(m_app.get_logger(), DNET_LOG_INFO,
+                    "Backend %s has moved from group %d to group %d",
+                    backend.get_key().c_str(), old_id, int(backend.get_stat().group));
+
+            it_old->second.remove_backend(backend);
```

Comment:
In this situation there's nothing special at this place. It will affect group status, it will change after the next recalculation.